kmod/src/tseq.c

```diff
@@ -46,14 +46,18 @@ static struct scoutfs_tseq_entry *tseq_rb_next(struct scoutfs_tseq_entry *ent)
 	return rb_entry(node, struct scoutfs_tseq_entry, node);
 }
 
+#ifdef RB_DECLARE_CALLBACKS_MAX
+static loff_t tseq_compute_total(struct scoutfs_tseq_entry *ent, int __attribute__((__unused__)) flag)
```

Comment:
This additional arg definitely needs to be used in the newer compute semantics.  It doesn't return the augmented value anymore, it updates the value itself and then can either return to short circuit or not.  Look at the _compute_max template in the rbtree_augmented.h header.  Off the top of my head, I imagine we'd leave this existing  _compute_total function unchanged and then add a _compute_and_set variant that checks the arg (bool exit, not int flags) to either set and return false or skip and return true.  If we have to have a KC macro anyway, we can just pass both functions in and use whichever is needed.

(and fwiw, using `__attribute__((attr))` is almost never the right thing in the kernel 'cause it supports other compilers.. there's usually a __attr equivalent.. __always_unused in this case (you often run into others like __packed, __weak, __deprecated, etc))

Reply:
Okay, I think I have this down now. We are _not_ looking to use a compute callback to calculate the max() of the augmented values or anything, but instead we want to look at and compute the total. So we do not have a use for the RB_DECLARE_CALLBACKS_MAX() macro - it simply defines an extra callback that handles the typical case where we're interested in the max() function of the node's and subtree's augmented value.

Incidentally, every single user in the kernel went to switch to this _MAX() variant afterwards. Which is why you don't find any more uses of the old case where we just need the RB_DECLARE_CALLBACKS() macro - we don't need the extra auto-generated callback function that it generates - we have our own.

In short, we'll continue to use RB_DECLARE_CALLBACKS() and we will now have 2 different tseq_compute_total variants. The old kernels will use the old version which just returns the new augmented computed value for a node (recursing).  The new kernels will use the new variant which updates the total if it changes, and if passed an `exit` flag (bool), will stop propagation by returning a boolean flag.